internal/response.go

```diff
@@ -6,113 +6,49 @@ package internal
-					}
-					app.logger.Debug("application not found, using default", zap.Any("application", arg.Value), zap.String("default", s.defaultApplication))
-				} else {
-					return nil, fmt.Errorf("application not found: %v", arg.Value)
-				}
-			}
-		case "id":
-			id, ok := arg.Value.(string)
-			if !ok {
-				return nil, fmt.Errorf("invalid argument for http response id, string expected, got %v", arg.Value)
-			}
+func NewResponse(msg spoe.Message) (*response, error) {
+	args := msg.Args.Map()
+	resp := response{}
+	var err error
 
-			txInterface, err := app.cache.Get(id)
-			if err != nil {
-				app.logger.Error("failed to get transaction from cache", zap.String("transaction_id", id), zap.String("error", err.Error()), zap.String("app", app.name))
-				break
-			}
+	resp.app, err = getString(args, "app")
+	if err != nil {
+		return nil, err
+	}
 
-			if tx, ok = txInterface.(types.Transaction); !ok {
-				app.logger.Error("Application cache is corrupted", zap.String("transaction_id", id), zap.String("app", app.name))
-				return nil, fmt.Errorf("application cache is corrupted")
-			}
-		case "version":
-			version, ok = arg.Value.(string)
-			if !ok {
-				app.logger.Error(fmt.Sprintf("invalid argument for http response version, string expected, got %v", arg.Value))
-				version = "1.1"
-			}
-		case "status":
-			status, ok = arg.Value.(int)
-			if !ok {
-				return nil, fmt.Errorf("invalid argument for http response status, int expected, got %v", arg.Value)
-			}
-		case "headers":
-			value, ok := arg.Value.(string)
-			if !ok {
-				app.logger.Error(fmt.Sprintf("invalid argument for http response headers, string expected, got %v", arg.Value))
-				value = ""
-			}
-			headers, err := s.readHeaders(value)
-			if err != nil {
-				return nil, err
-			}
-			for key, values := range headers {
-				for _, v := range values {
-					tx.AddResponseHeader(key, v)
-				}
-			}
-		case "body":
-			body, ok := arg.Value.([]byte)
-			if !ok {
-				return nil, fmt.Errorf("invalid argument for http response body, []byte expected, got %v", arg.Value)
-			}
-			it, _, err := tx.WriteRequestBody(body)
-			if err != nil {
-				return nil, err
-			}
-			if it != nil {
-				return s.processInterruption(it, hit), nil
-			}
-		default:
-			app.logger.Warn(fmt.Sprintf("invalid message on the http response, name: %s, value: %s", arg.Name, arg.Value))
-		}
+	resp.id, err = getString(args, "id")
+	if err != nil {
+		return nil, err
 	}
 
-	if it := tx.ProcessResponseHeaders(status, "HTTP/"+version); it != nil {
-		return s.processInterruption(it, hit), nil
+	resp.version, err = getString(args, "version")
+	if err != nil {
+		fmt.Println(err.Error())
 	}
-	it, err := tx.ProcessResponseBody()
+
+	resp.status, err = getInt(args, "status")
 	if err != nil {
 		return nil, err
 	}
-	if it != nil {
-		return s.processInterruption(it, hit), nil
+
+	resp.headers, err = getString(args, "headers")
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
Instead of fmt.Println could we inject the logger from the app?

Reply:
Let's deal with logging in  #70.